pkg/innerring/settlement.go

```diff
@@ -204,6 +217,17 @@ func (s settlementDeps) ResolveKey(ni common.NodeInfo) (*owner.ID, error) {
 }
 
 func (s settlementDeps) Transfer(sender, recipient *owner.ID, amount *big.Int, details []byte) {
+	var callerContext string
+
+	switch s.settlementCtx {
```

Comment:
Oh, u've done a great job! But maybe it's easeir to have `name string` field in `settlementDeps` struct? Minus type, minus enum, minus switch.

Reply:
thought about that but then did not want to have string here

but ok, agree that it is easier, fixed